src/Arpack.jl

```diff
@@ -339,8 +339,9 @@ function _svds(X; nsv::Int = 6, ritzvec::Bool = true, tol::Float64 = 0.0, maxite
     end
     m, n = size(X)
     otype = eltype(X)
-    if length(v0) ∉ [0,n]
-        throw(DimensionMismatch("length of v0, the guess for the starting right Krylov vector, must be 0, or $n, got $(length(v0))"))
+    if length(v0) ∉ [0,min(m, n)]
```

Comment:
```suggestion
    if length(v0) ∉ (0, min(m, n))
```
to avoid the vector allocation now that you are here anyway.